### PR TITLE
DHFPROD-2546: Can now install/upgrade DHF via an executable jar

### PR DESCRIPTION
**Pull request by rjrudin**
See the comments in Installer for how to build and run it. 

Also removed the "main" method from ApplicationConfig as I don't believe there's any use case for it.

### CODE REVIEW

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    /**
+     * If a user already exists, we don't want to overwrite its password.
+     */
+    protected void removePasswordsFromUserFiles(File projectDir, HubConfigImpl hubConfig) {
+        ResourcesFragment existingUsers = new UserManager(hubConfig.getManageClient()).getAsXml();
+        File dir = new File(projectDir, "src/main/hub-internal-config");
+        if (dir.exists()) {
+            final Map<String, String> tokens = hubConfig.getAppConfig().getCustomTokens();
+
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java]:
```suggestion
if (tokens == null) {
   logger.warn(...)
   return;
}
```

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            File usersDir = new ConfigDir(dir).getUsersDir();
+            if (usersDir != null && usersDir.exists()) {
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java]: usersDir can be null ?

Comment (rjrudin) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java]: Good point - I'll remove that

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java (hunk 1 above)
Comment (rjrudin) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/cli/Installer.java]: The token map wouldn't be null, as it's eagerly instantiated by AppConfig. AppConfig could be null, which would mean that the hubConfig.refresh call failed, and nothing's going to work, in which case an exception will naturally occur.

**Pull request merged**